Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp

```diff
@@ -88,7 +88,12 @@ static UnixFileDescriptor findDevice(GError** error)
             continue;
 
         fd = UnixFileDescriptor { open(device->nodes[DRM_NODE_PRIMARY], O_RDWR | O_CLOEXEC), UnixFileDescriptor::Adopt };
+
         if (fd)
+            continue;
+
+        WPE::DRM::UniquePtr<drmModeRes> resources(drmModeGetResources(fd.value()));
+        if (resources)
```

Comment:
You are continuing the loop when the fd is valid. I agree we could check that the device is actually a valid DRM device we can use for modesetting. Note that drmModeGetResources returns nullptr on error, not because the device doesn't have resources, I guess it's a permission error in you case. So, we could also check that indeed has resources with something like:

```
WPE::DRM::UniquePtr<drmModeRes> resources(drmModeGetResources(fd.value()));
if (resources && resources->count_crtcs && resources->count_connectors && resources->count_encoders)
```

But again, let's move this to its own PR, please use https://bugs.webkit.org/show_bug.cgi?id=268535

Reply:
I am seeing typos in this code (probably during the clean-up). I will separate the this in a single ticket as you suggest.